packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    CarouselHandledProps,
+    CarouselProps,
+    CarouselUnhandledProps,
+    Slide,
+    SlideTheme,
+} from "./carousel.props";
+import { Flipper, FlipperDirection } from "../flipper";
+import { Tabs } from "@microsoft/fast-components-react-base";
+import { TabsClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { get } from "lodash-es";
+
+/**
+ * The carousel state interface
+ */
+export interface CarouselState {
+    /**
+     * Holds the active tab id to share with other controls
+     */
+    activeId: string;
+}
+
+class Carousel extends Foundation<
+    CarouselHandledProps,
+    CarouselUnhandledProps,
+    CarouselState
+> {
+    public static displayName: string = "Carousel";
+
+    protected handledProps: HandledProps<CarouselHandledProps> = {
+        managedClasses: void 0,
+        label: void 0,
+        activeId: void 0,
+        items: void 0,
+    };
+
+    private slides: Slide[];
+
+    /**
+     * Define constructor
+     */
+    constructor(props: CarouselProps) {
+        super(props);
+
+        if (this.props.items) {
+            this.state = {
+                activeId:
+                    typeof this.props.activeId === "string"
+                        ? this.props.activeId
+                        : this.props.items.length > 0
+                            ? get(this.props.items[0], "id")
+                            : "",
+            };
+
+            this.updateSlides();
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidUpdate(): void {
+        if (this.slides !== this.props.items) {
+            this.updateSlides();
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div {...this.unhandledProps()} className={this.generateClassNames()}>
+                <Flipper
+                    direction={FlipperDirection.previous}
+                    onClick={this.handlePreviousInvoke}
+                    className={get(
+                        this.props,
+                        "managedClasses.carousel_flipperPrevious",
+                        ""
+                    )}
+                />
```

Comment:
I pulled this down and am noticing that these flippers always exist and they are also always visible. As a sighted keyboard user, I am confused as to why I can't access the flippers. I suggest that we consider defaulting these to only show on hover. The question becomes, "what happens on mobile?" We can consider adding swipe out of the box, or we could also consider exposing a prop to have the flippers always be visible.

Reply:
If you review the notes, I left details on these decisions. Since there is no design or spec as of yet, I started with the most simple version of the carousel. So I can obviously change the functionality of how the flippers act, but I wanted to open up the conversation.

Reply:
Sure - I wanted to start that conversation here; I likely could have led with that caveat. 

In the scenario of a tabpanel, I think flippers function as an ease-of-use for mouse users. Keyboard users should likely never be exposed to these as they act as duplicate navigation. With that in mind, we should consider only showing them on hover, which is a typical solution on hover. 

Open to suggestions on how we handle mobile scenarios for this.

Reply:
May want to lean on design for some input on this... would they even be open to flippers on mobile? If not, then swipe is probably our only option.

Reply:
Changed to visible on hover for now. Not sure how we want to handle mobile-side. May want weigh-in from design.